gatorgrouper/tests/test_read_student_file.py

```diff
@@ -17,12 +18,43 @@ def generate_csv(tmpdir_factory):
         writer = csv.DictWriter(csvfile, fieldnames=headers)
         writer.writeheader()
         writer.writerow(
-            {"NAME": "delgrecoj", "Q1": True, "Q2": True, "Q3": False, "Q4": True}
+            {
+                "NAME": "delgrecoj",
+                "Q1": "True",
+                "Q2": "True",
+                "Q3": "False",
+                "Q4": "True",
+            }
         )
     return str(fn)
 
 
+@pytest.fixture(scope="session")
+def generate_csv_no_header(tmpdir_factory):
+    """ Generate a tempory sample csv """
+    fn = tmpdir_factory.mktemp("data").join("csvNg1.csv")
+    df_list = {
+        "NAME": ["delgrecoj", "delgrecoj2"],
+        "Q1": ["True", "True"],
+        "Q2": ["True", "True"],
+        "Q3": ["False", "True"],
+        "Q4": ["True", "True"],
+    }
+    df = pd.DataFrame(df_list)
```

Comment:
There is no need to use `pandas` here... why not just write the `csv` manually or with a `csv.DictWriter`? No need to add a dependency for something so simple.

Reply:
Hi @Michionlion, this is because `csv.DictWriter` can only accept the parameter of dictionary which means header is required as the key in dictionary. I did a lot of research and I still did not find the way to generate csv without header by using csv package. However, by using `pandas` package, I can set `header=False` when I am implementing `to_csv` function. Which will generate a csv without header.

Reply:
If we just need to generate CSV files for test cases (i.e. as long as we don't have to do anything with properly escaping commas in the strings, etc), then I believe we could make a headerless CSV file by doing
```
data = [
    # optionally include headers as the first entry
    ["delgrecoj", "True", "True", "False", "True"],
    ["delgrecoj2", "True", "True", "True", "True"]
]
csv_string = ""
for entry in data:
     csv_string += ",".join(entry) +"\r\n"

# write csv_string to the output file
...
```

Reply:
I have tried many different ways including this. You can find that in one of the commit history. It turns out that they would not generate the non-header-csv properly. For some reason it will always recognize the first line as the header, and I finally get it work by using `pandas`.

Reply:
print out the actual string generated by `pandas`, and use that.

Reply:
@Michionlion I will check that

Reply:
I have tried and it does not work. The only way to generate csv with `tmpdir_factory` besides using `pandas is to use `csv.DictWriter(csvfile, fieldnames=headers)` and in this situation the header will always be recognized by `read_student_file` since `csv` does not have the feature to set `header=False`.

Reply:
Actually, nvm, I think I figured it out. I thought directwrite is the only way to generate csv.